### DISCUSSION TITLE
pocketbase stuck and gives no response

### DISCUSSION THREAD

**Discussion by Oggy4th** [pocketbase/pocketbase#4078] [Q&A]
Hello, there

My pocketbase suddenly stuck and gave no response.

![image](https://github.com/pocketbase/pocketbase/assets/44497354/071150f3-cac9-4113-9ca9-d85c2b67e08e)

Everything is loading but there is no response.

Is it my coding error or something else?

I wrote all functions elapsed time in error.log file.

```logtalk
2024/01/04 10:23:15 BranchCheck elapsed time: 839.501µs
2024/01/04 10:23:19 BranchCheck elapsed time: 995.599µs
2024/01/04 10:23:20 BranchCheck elapsed time: 1.033817ms
2024/01/04 10:23:32 BranchCheck elapsed time: 1.621676ms
2024/01/04 10:23:38 BranchCheck elapsed time: 1.201994ms
2024/01/04 10:24:04 BranchCheck elapsed time: 675.112µs
2024/01/04 10:24:04 http: TLS handshake error from 202.180.219.250:51990: context canceled
2024/01/04 10:24:04 http: TLS handshake error from 202.180.219.230:61701: context canceled
2024/01/04 10:24:04 http: TLS handshake error from 202.180.219.254:55755: context canceled
2024/01/04 10:24:04 http: TLS handshake error from 202.180.219.25:50805: context canceled
2024/01/04 10:24:04 http: TLS handshake error from 202.180.219.23:2262: context canceled
2024/01/04 10:24:04 http: TLS handshake error from 202.180.219.25:57252: context canceled
2024/01/04 10:24:04 http: TLS handshake error from 202.180.219.27:50530: context canceled
2024/01/04 10:24:07 PostpaidSession elapsed time: 50.130206455s
2024/01/04 10:24:07 Session elapsed time: 2m49.850030657s
2024/01/04 10:24:07 OtaSession elapsed time: 5m16.499542733s
2024/01/04 10:24:07 Session elapsed time: 17.183479681s
2024/01/04 10:24:07 OtaSession elapsed time: 2m47.957000441s
2024/01/04 10:24:07 OtaSession elapsed time: 4m16.664284245s
2024/01/04 10:24:07 Session elapsed time: 1m54.024155686s
2024/01/04 10:24:07 Session elapsed time: 2m48.273672574s
2024/01/04 10:24:07 Session elapsed time: 48.570260362s
```

**Comment by ganigeorgiev** [comment]
I don't know how to reproduce it and I suspect that it is most like an error in your custom hooks code (considering that all errors are coming from the custom `/api/user/session` and `/api/ota/session` routes).

Another thing that can cause the `database is closed` error is if you have an external application that misbehave and operates on the same SQLite db file (Litestream, Marmot, etc.).

I remember you from posting similar errors in the past in https://github.com/pocketbase/pocketbase/discussions/3755, https://github.com/pocketbase/pocketbase/discussions/3166, https://github.com/pocketbase/pocketbase/discussions/3093, https://github.com/pocketbase/pocketbase/discussions/3067, etc. but please understand that without a **minimal** reproducible example showing what your custom code is doing I don't know how to help you.

It is possible also to be some obscure bug in PocketBase, I'm not ruling that out, but so far I haven't stumbled on it during my local development with Presentator or on the live demo, nor I think anyone else reported similar issue.

**Comment by CovertCode** [comment]
Are you updating a collection via hook?

**Reply by Oggy4th** [reply]
Yes, I am. I am using multiple hooks.

**Reply by ganigeorgiev** [reply]
@Oggy4th Without a code sample, there are too many things to try to guess what could have gone wrong.

Double check your transactions (if you have any) and whether you are passing the correct Dao. For example, if you have a db write with `app.Dao()` from within a transaction then this most likely will end up in a deadlock because the transaction Dao will not be able to proceed further since the `app.Dao()` write will wait for the transaction to complete.

If you are not able to identify what is causing it to "stuck", please share your custom code or create a **minimal** reproducible example illustrating the problem.

**Comment by jonshipman** [comment]
Just throwing this out there, your DB might be corrupt? Can you reproduce with a clean pb_data folder?
